src/QueryBuilder/Condition/Builder/NotBuilder.php

```diff
@@ -36,19 +53,66 @@ public function __construct(private readonly QueryBuilderInterface $queryBuilder
     {
-        $operand = $expression->condition;
+        $condition = is_array($expression->condition)
+            ? $this->queryBuilder->createConditionFromArray($expression->condition)
+            : $expression->condition;
 
-        if ($operand === '') {
+        if ($condition === null || $condition === '') {
             return '';
         }
 
-        $expressionValue = $this->queryBuilder->buildCondition($operand, $params);
+        if ($condition instanceof ConditionInterface) {
+            $negatedCondition = $this->tryCreateNegatedCondition($condition);
+            if ($negatedCondition !== null) {
+                return $this->queryBuilder->buildCondition($negatedCondition, $params);
+            }
+        }
 
-        return "{$this->getNegationOperator()} ($expressionValue)";
+        $sql = $this->queryBuilder->buildCondition($condition, $params);
+        return "NOT ($sql)";
     }
 
-    protected function getNegationOperator(): string
+    protected function tryCreateNegatedCondition(ConditionInterface $condition): array|string|ExpressionInterface|null
     {
-        return 'NOT';
+        return match ($condition::class) {
+            LessThan::class => new GreaterThanOrEqual($condition->column, $condition->value),
+            LessThanOrEqual::class => new GreaterThan($condition->column, $condition->value),
+            GreaterThan::class => new LessThanOrEqual($condition->column, $condition->value),
+            GreaterThanOrEqual::class => new LessThan($condition->column, $condition->value),
+            In::class => new NotIn($condition->column, $condition->values),
+            NotIn::class => new In($condition->column, $condition->values),
+            Between::class => new NotBetween(
+                $condition->column,
+                $condition->intervalStart,
+                $condition->intervalEnd,
+            ),
+            NotBetween::class => new Between(
+                $condition->column,
+                $condition->intervalStart,
+                $condition->intervalEnd,
+            ),
+            Equals::class => new NotEquals($condition->column, $condition->value),
+            NotEquals::class => new Equals($condition->column, $condition->value),
+            Exists::class => new NotExists($condition->query),
+            NotExists::class => new Exists($condition->query),
+            Like::class => new NotLike(
+                $condition->column,
+                $condition->value,
+                $condition->caseSensitive,
+                $condition->escape,
+                $condition->mode,
+                $condition->conjunction,
+            ),
+            NotLike::class => new Like(
+                $condition->column,
+                $condition->value,
+                $condition->caseSensitive,
+                $condition->escape,
+                $condition->mode,
+                $condition->conjunction,
+            ),
+            Not::class => $condition->condition,
```

Comment:
What if `$condition->condition === null`?

Reply:
Good catch! I fixed it. Review it.